### PR TITLE
Retrieve proper key for Adyen payments

### PR DESCRIPTION
**Pull request by mionescu95**
## JIRA
[MOB-4473](https://karhoo.atlassian.net/browse/MOB-4473)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/adyen/java/com/karhoo/uisdk/AdyenPaymentPresenter.kt [hunk 1]
```diff
@@ -85,18 +85,23 @@ class AdyenPaymentPresenter(
             Environment.TEST
         }
 
-        val googlePayConfig = GooglePayConfiguration.Builder(context, clientKey)
+        val key: String = if(userStore.paymentProvider?.provider?.version.equals("v68", ignoreCase = true))
```
Comment (dianakarh) [inline uisdk/src/adyen/java/com/karhoo/uisdk/AdyenPaymentPresenter.kt:88]: The version should be >= 68. Let's not get stuck on v68 only

File: uisdk/src/adyen/java/com/karhoo/uisdk/AdyenPaymentPresenter.kt [hunk 2]
```diff
@@ -205,29 +210,34 @@ class AdyenPaymentPresenter(
...
     override fun sdkInit(quote: Quote?, locale: Locale?) {
         this.quote = quote
-        paymentsService.getAdyenPublicKey().execute { result ->
-            when (result) {
-                is Resource.Success -> {
-                    result.data.let {
-                        adyenKey = it.publicKey
-                        getPaymentMethods(locale)
+
+        if(!userStore.paymentProvider?.provider?.version.equals("v68", ignoreCase = true)) {
```
Comment (dianakarh) [inline uisdk/src/adyen/java/com/karhoo/uisdk/AdyenPaymentPresenter.kt:214]: The rule is >= v69 => clientKey, else => public key

**Review by dianakarh — CHANGES_REQUESTED** [review]
The key selection logic needs updating.
if adyen v >= 68 => clientKey
else => publicKey

**Review by dianakarh — APPROVED** [review]
Unblocking the PR before going OOO. @mateid-kr please review this when done.

**Review by mateid-kr — APPROVED** [review]

**Pull request merged**